android/src/com/mapswithme/maps/bookmarks/BookmarksListFragment.java

```diff
@@ -739,26 +669,131 @@ public boolean onOptionsItemSelected(MenuItem item)
-          .listener(this::onListMoreMenuItemClick)
-          .build();
-
-      @BookmarkManager.SortingType int[] types = getAvailableSortingTypes();
-      Menu moreMenu = bs.getMenu();
-      moreMenu.findItem(R.id.sort).setVisible(types.length > 0 && !isEmpty());
-      moreMenu.findItem(R.id.delete_category).setVisible(!isLastOwnedCategory());
-      moreMenu.findItem(R.id.share_category).setVisible(!isEmpty());
-
-      BottomSheetHelper.tint(bs);
-      bs.show();
+      new MenuBottomSheetFragment(mCategoryDataSource.getData().getName(), getOptionsMenuItems())
+              .show(getParentFragmentManager(), "optionsBottomSheet");
       return true;
     }
 
     return super.onOptionsItemSelected(item);
   }
 
+  private void onShareActionSelected()
+  {
+    BookmarkInfo info = (BookmarkInfo) getBookmarkListAdapter().getItem(mSelectedPosition);
+    SharingUtils.shareBookmark(requireContext(), info);
+  }
+
+  private void onEditActionSelected()
+  {
+    BookmarkListAdapter adapter = getBookmarkListAdapter();
+    BookmarkInfo info = (BookmarkInfo) adapter.getItem(mSelectedPosition);
+    EditBookmarkFragment.editBookmark(
+        info.getCategoryId(), info.getBookmarkId(), requireActivity(), getChildFragmentManager(),
+        (bookmarkId, movedFromCategory) ->
+        {
+          if (movedFromCategory)
+            resetSearchAndSort();
+          else
+            adapter.notifyDataSetChanged();
+        });
+  }
+
+  private void onDeleteActionSelected()
+  {
+    BookmarkListAdapter adapter = getBookmarkListAdapter();
+    BookmarkInfo info = (BookmarkInfo) getBookmarkListAdapter().getItem(mSelectedPosition);
+    adapter.onDelete(mSelectedPosition);
+    BookmarkManager.INSTANCE.deleteBookmark(info.getBookmarkId());
+    adapter.notifyDataSetChanged();
+    if (mSearchMode)
+      mNeedUpdateSorting = true;
+    updateSearchVisibility();
+    updateRecyclerVisibility();
+  }
+
+  private void onSortOptionSelected()
+  {
+    ChooseBookmarksSortingTypeFragment.chooseSortingType(getAvailableSortingTypes(),
+        getLastSortingType(), requireActivity(), getChildFragmentManager());
+  }
+
+  private void onShareOptionSelected()
+  {
+    long catId = mCategoryDataSource.getData().getId();
+    BookmarksSharingHelper.INSTANCE.prepareBookmarkCategoryForSharing(requireActivity(), catId);
+  }
+
+  private void onSettingsOptionSelected()
+  {
+    BookmarkCategorySettingsActivity.startForResult(this, mCategoryDataSource.getData());
+  }
+
+  private void onDeleteOptionSelected()
+  {
+    requireActivity().setResult(Activity.RESULT_OK);
+    requireActivity().finish();
+  }
+
+  private ArrayList<MenuBottomSheetItem> getOptionsMenuItems()
+  {
+    @BookmarkManager.SortingType int[] types = getAvailableSortingTypes();
+    ArrayList<MenuBottomSheetItem> items = new ArrayList<>();
+    if (!isEmpty())
+    {
+      if (types.length > 0)
+      {
+        items.add(new MenuBottomSheetItem(
```

Comment:
nit: items.add here and in other places can be made as one-liners to read functions easier.